### PR TITLE
feat: add flag for controlling chat participant and disable it when release rc/beta

### PR DESCRIPTION
**Pull request by Alive-Fish**
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/26624499

Add `TEAMSFX_CHAT_PARTICIPANT` flag and set it to `true` in the code. When releasing `rc/stable`, change it to false for hiding other entries of chat participant.

Set `contributes.chatParticipants` to `[]` to hide chat participant when releasing `rc/stable`.

Use `fx-extension.isChatParticipantEnabled` in `package.json` to control the welcome page content and webview content.

Test in https://github.com/OfficeDev/TeamsFx/actions/runs/8462480742.

### CODE REVIEW

File: packages/vscode-extension/src/extension.ts [hunk 1]
```diff
@@ -126,6 +132,15 @@ export async function activate(context: vscode.ExtensionContext) {
   // UI is ready to show & interact
   await vscode.commands.executeCommand("setContext", "fx-extension.isTeamsFx", isTeamsFxProject);
 
+  // control whether to show chat participant entries
+  await vscode.commands.executeCommand(
+    "setContext",
+    "fx-extension.isChatParticipantEnabled",
```
Comment (tecton) [inline packages/vscode-extension/src/extension.ts:138]: Will we use this variable in `package.json`?

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -188,6 +188,23 @@ jobs:
           git add packages/fx-core/src/common/m365/serviceConstant.ts
           git commit -m "build: replace sideloading placeholders"
 
+      - name: disable chat participant in package.json
```
Comment (tecton) [inline .github/workflows/cd.yml:191]: We could also determine whether the extension is prerelease based on fx-core version and vscode-extension version. Is that one more convenient?

File: packages/vscode-extension/src/extension.ts (hunk 1 above)
Comment (Alive-Fish) [reply packages/vscode-extension/src/extension.ts:138]: Yes

File: .github/workflows/cd.yml (hunk 2 above)
Comment (tecton) [reply .github/workflows/cd.yml:190]: Confirmed that we also need to remove contributes.chatparticipants in package.json from script.

**Review by tecton — APPROVED** [review]

**Review by wenytang-ms — APPROVED** [review]

**Pull request merged**